CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.cc

```diff
@@ -21,243 +21,85 @@ CaloTPGTranscoderULUT::CaloTPGTranscoderULUT(const std::string& compressionFile,
                                                   compressionFile_(compressionFile),
                                                   decompressionFile_(decompressionFile)
 {
-  for (int i = 0; i < NOUTLUTS; i++) outputLUT_[i] = 0;
+  outputLUT_.clear();
 }
 
 CaloTPGTranscoderULUT::~CaloTPGTranscoderULUT() {
-  for (int i = 0; i < NOUTLUTS; i++) {
-    if (outputLUT_[i] != 0) delete [] outputLUT_[i];
-  }
 }
 
 void CaloTPGTranscoderULUT::loadHCALCompress(HcalLutMetadata const& lutMetadata,
                                              HcalTrigTowerGeometry const& theTrigTowerGeometry) {
-// Initialize analytical compression LUT's here
-   // TODO cms::Error log
-  if (OUTPUT_LUT_SIZE != (unsigned int) 0x400) std::cout << "Error: Analytic compression expects 10-bit LUT; found LUT with " << OUTPUT_LUT_SIZE << " entries instead" << std::endl;
+    // Initialize analytical compression LUT's here
+    // TODO cms::Error log
+    if (OUTPUT_LUT_SIZE != (unsigned int) 0x400) std::cout << "Error: Analytic compression expects 10-bit LUT; found LUT with " << OUTPUT_LUT_SIZE << " entries instead" << std::endl;
 
-  std::vector<unsigned int> analyticalLUT(OUTPUT_LUT_SIZE, 0);
-  std::vector<unsigned int> identityLUT(OUTPUT_LUT_SIZE, 0);
+    std::vector<unsigned int> analyticalLUT(OUTPUT_LUT_SIZE, 0);
+    std::vector<unsigned int> identityLUT(OUTPUT_LUT_SIZE, 0);
 
-  // Compute compression LUT
-  for (unsigned int i=0; i < OUTPUT_LUT_SIZE; i++) {
+    // Compute compression LUT
+    for (unsigned int i=0; i < OUTPUT_LUT_SIZE; i++) {
 	analyticalLUT[i] = (unsigned int)(sqrt(14.94*log(1.+i/14.94)*i) + 0.5);
 	identityLUT[i] = min(i,0xffu);
-  }
+    }
```

Comment:
Avoid magic numbers. We have no idea of their meaning.
Still, same comment as above applies.